File: Samples/BasicApplication/BasicApplication/BasicApplication.Script/Northwind/Customer/CustomerEditor.cs
namespace BasicApplication.Northwind
{
    using jQueryApi;
    using Serenity;

    [Editor]
    public class CustomerEditor : LookupEditorBase<object, CustomerRow>
    {
        public CustomerEditor(jQueryObject select)
            : base(select, null)
        {
        }

        protected override string GetLookupKey()
        {
            return "Northwind.Customer";
        }
    }
}

File: common-features/src/demo.northwind/Modules/Customer/CustomerEditor.ts
import { EditorProps, LookupEditorBase, LookupEditorOptions } from "@serenity-is/corelib";
import { CustomerRow } from "../ServerTypes/Demo";
import { nsDemoNorthwind } from "../ServerTypes/Namespaces";

export class CustomerEditor<P extends LookupEditorOptions = LookupEditorOptions> extends LookupEditorBase<P, CustomerRow> {
    static override[Symbol.typeInfo] = this.registerEditor(nsDemoNorthwind);

    constructor(props: EditorProps<P>) {
        super({ async: true, ...props });
    }

    protected override getLookupKey() {
        return 'Northwind.Customer';
    }

    protected override getItemText(item, lookup) {
        return super.getItemText(item, lookup) + ' [' + item.CustomerID + ']';
    }
}
